### PR TITLE
Add tests for `$$ROOT`

### LINKED ISSUE #1992: Implement ROOT aggregation system variable
### What should be done?

* https://www.mongodb.com/docs/v6.0/meta/aggregation-quick-reference/#aggregation-variables
* https://www.mongodb.com/docs/v6.0/reference/aggregation-variables/#mongodb-variable-variable.ROOT

`$$ROOT` expression variable returns the document at its root of the current stage (not the initial input to the pipeline), see example below.

```
test> db.Nulls.aggregate([{$group: { "_id": "$$ROOT" } }])
[ { _id: { _id: 'null', v: null } } ]
```

See also skipped compat tests in the codebase.

### Where?
https://github.com/FerretDB/FerretDB/tree/main/internal/handlers/common/aggregations

### Definition of Done
- create `root.go` that implements `$$ROOT` variable;
- use similar pattern as `Stage`,`Accumulator`,`Operator` for creating `Expression` variable;
- integration/compatibility tests added/updated;
- spot refactorings done;
- user documentation updated or an issue to create documentation created;
- unskip compat tests with issue number

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1992.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/aggregate_variables_compat_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func TestAggregateVariablesCompatRoot(t *testing.T) {
 				bson.D{{"$addFields", bson.D{{"field", "$$ROOT"}}}},
 			},
 			failsForSQLite: "https://github.com/FerretDB/FerretDB/issues/3148",
-			skip:           "https://github.com/FerretDB/FerretDB/issues/1992",
+			skip:           "https://github.com/FerretDB/FerretDB/issues/1413",
```
Comment (chilagrow) [inline integration/aggregate_variables_compat_test.go:36]: Actually `$set` and `$addFields` do not support expression yet. So let's not introduce that until they are implemented.

File: integration/aggregate_variables_compat_test.go [hunk 2]
```diff
@@ -44,6 +44,16 @@ func TestAggregateVariablesCompatRoot(t *testing.T) {
 			failsForSQLite: "https://github.com/FerretDB/FerretDB/issues/3148",
 			skip:           "https://github.com/FerretDB/FerretDB/issues/1992",
 		},
+		"GroupIDTwice": {
+			pipeline: bson.A{
+				bson.D{{"$sort", bson.D{{"_id", 1}}}},
+				bson.D{{"$group", bson.D{{"_id", "$$ROOT"}}}},
+				bson.D{{"$group", bson.D{{"_id", "$$ROOT"}}}},
+				bson.D{{"$sort", bson.D{{"_id", 1}}}},
+			},
```
Comment (chilagrow) [inline integration/aggregate_variables_compat_test.go:53]: $$ROOT uses document of its own pipeline stage, not the initial pipeline stage. It's rather strange example, found no other stage to demonstrate it. `$addFields` doesn't implement expression yet.
```
test> db.coll.insertOne({_id:1,v:1})                                                                                                                                                                        
{ acknowledged: true, insertedId: 1 }  
test> db.coll.aggregate([{ $group: { "_id": "$$ROOT" } }, { $group: { "_id": "$$ROOT" } }])           
[                                                                                                     
  { _id: { _id: { _id: 1, v: 1 } } }                                                                  
] 
```

**Review by rumyantseva — APPROVED** [review]
Great! Thanks for the additional examples!

**Review by noisersup — APPROVED** [review]
LGTM

**Pull request merged**